docs/DevGuide/TagsTypelists.md

```diff
@@ -0,0 +1,190 @@
+to tags, so we can no longer use `std::map<std::string, double>` as our
+container. We need a container that is *heterogeneous*, that is, is capable
+of holding objects of different types. In C++, this is done through templates.
+
+A well-documented example of a fixed-size heterogeneous container of types is
+[std::tuple](https://en.cppreference.com/w/cpp/utility/tuple):
+
+~~~{.c}
+std::tuple<double, size_t, bool> sophomore_databox =
+  std::make_tuple(1.2, 8, true);
+~~~
+
+The quantities are obtained using
+[std::get](https://en.cppreference.com/w/cpp/utility/tuple/get):
+
+~~~{.c}
+bool_quantity = std::get<bool>(sophomore_databox);
+
+// value of obtained is `true`
+~~~
+
+Note that because our keys are now types as opposed to values, the way by which
+we index into our container has changed. If we were still using a `std::map`,
+we would index into our container like this:
+
+~~~{.c}
+const bool bool_quantity = true;
+
+std::map<std::string, bool> naive_databox;
+naive_databox["Bool"] = bool_quantity;
+
+return naive_databox["Bool"];
+~~~
+
+SpECTRE's `tmpl::list`s are not `std::tuples`, but rather a wrapped version of
+`brigand::list`s provided by
+[Brigand](https://github.com/edouarda/brigand/wiki/Introduction). We encourage
+you to look at their brief introduction in order for you to familiarize yourself
+with compile time containers and how their usage differs from that of runtime
+containers.
+
+# Type Lists in SpECTRE {#documentation_for_typelists_in_spectre}
+
+## tmpl::list
+
+This non-associative container only holds types. That is, there is no variable
+runtime data associated with it like there is for `std::tuple`, which is a
+container associating types with values. They are useful in situations when one
+is working with multiple tags at once.
+
+## tuples::TaggedTuple
+
+The spirit of `tuples::TaggedTuple` is a `std::tuple` where the keys are tags:
+~~~{.c}
+struct Pressure {
+  using type = double;
+};
+struct Density {
+  using type = double;
+};
+
+tuples::TaggedTuple<Pressure, Density> junior_databox{1.2, 0.5};
+return tuples::get<Pressure>(junior_databox);
+~~~
+
+You'll find `TaggedTuple`s are often used in AnalyticSolutions, among other
+places. Here is a simplified example of its use:
+
+~~~{.c}
+tuples::TaggedTuple<Pressure> pressure_variable(
+  const tnsr::I<double, 3>& /*x*/, double /*t*/) const
+  noexcept {
+    const double pressure = 1.2;
+    return {Scalar<double>(pressure)};
+  }
+~~~
+
+## Variables
+
+The `Variables` data structure is a more restricted form of `TaggedTuple` where
+the tags must be `SimpleTag`s, and where the types must be `Tensor`s. They are
+typically not returned, and are used in intermediate calculations when it is
+desirable to allocate memory only once.
+
+## DataBox
+
+The simplified DataBox introduction, while simplified, is still extensive. You
+will find the introduction in the \ref databox_concepts "DataBox" section of
+the Dev Guide.
```

Comment:
I would rephrase this, to make it more clear and concise:
A simplified, but still extensive introduction to SpECTRE's databox can be found in the \ref databox_concepts "DataBox" section of the Dev Guide.